llvm/lib/CodeGen/GlobalISel/CombinerHelper.cpp

```diff
@@ -7400,3 +7400,23 @@ bool CombinerHelper::matchZextOfTrunc(const MachineOperand &MO,
 
   return false;
 }
+
+bool CombinerHelper::matchNonNegZext(const MachineOperand &MO,
+                                     BuildFnTy &MatchInfo) {
+  GZext *Zext = cast<GZext>(getDefIgnoringCopies(MO.getReg(), MRI));
```

Comment:
Can this just to get def? It shouldn't have to worry about intervening copies (I think we have too many uses of getDefIgnoringCopies)